### PR TITLE
Implement Line chart

### LINKED ISSUE #211: Report engine - Implement Line and combine (line + bar) chart
Like the bar chart or the pie chart, we need to implement the line chart
In the same time, the combine chart (line + bar) will be implemented.

### PR DESCRIPTION
**Pull request by Alteoris**
Implement Line chart with customization

close #211

### CODE REVIEW

**Review by HisCodeness — APPROVED** [review]

File: OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/AxisModel.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    /// </summary>
+    public class AxisModel
+    {
+        /// <summary>
+        /// Title
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// Title color
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/AxisModel.cs:14]: Could you add a sample value to detail if we must provide the hexa code with or without # char

File: OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/SerieModel.cs [hunk 2]
```diff
@@ -24,6 +25,12 @@ public class SerieModel
         /// </summary>
         public string DataLabelColor { get; set; }
 
+        /// <summary>
+        /// Format de rendu des labels
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/SerieModel.cs:29]: In english ?

File: OpenXMLSDK.Engine/ReportEngine/DataContext/ContextModel.cs [hunk 3]
```diff
@@ -127,8 +127,6 @@ public void ReplaceItem(BarModel element, IFormatProvider formatProvider)
                 element.FontName = ReplaceText(element.FontName, formatProvider);
             if (!string.IsNullOrEmpty(element.FontColor))
                 element.FontColor = ReplaceText(element.FontColor, formatProvider);
-            if (!string.IsNullOrEmpty(element.MajorGridlinesColor))
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/ReportEngine/DataContext/ContextModel.cs:130]: Why did you remove this condition ?

File: OpenXMLSDK.Engine/Word/Extensions/ScalingModelExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace OpenXMLSDK.Engine.Word.Extensions
+{
+    public static class ScalingModelExtensions
+    {
+        /// <summary>
+        /// Construct a DocumentFormat.OpenXml.Drawing.Charts.Scaling object
+        /// </summary>
+        /// <returns></returns>
+        public static Scaling GetScaling(this ScalingModel model)
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/Word/Extensions/ScalingModelExtensions.cs:14]: What the behaviour if the model is null ?

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        /// </summary>
+        public bool DeleteAxis { get; set; }
+
+        /// <summary>
+        /// Indicate if the curve sepparating axis from graph must be deleted
+        /// </summary>
+        public bool ShowAxisCurve { get; set; }
+
+        /// <summary>
+        /// Define the color of the curve sepparating axis from graph
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs:29]: Could you add an example with expected value (with or without # char)

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        /// </summary>
+        public string AxisCurveColor { get; set; }
+
+        /// <summary>
+        /// Indicate if we show major grid lines
+        /// </summary>
+        public bool ShowMajorGridlines { get; set; }
+
+        /// <summary>
+        /// Indicate the color of major grid lines
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs:39]: Could you add an example with expected value (with or without # char)

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    /// </summary>
+    public class ChartAxisModel
+    {
+        /// <summary>
+        /// Title
+        /// </summary>
+        public string Title { get; set; }
+
+        /// <summary>
+        /// Title color
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartAxisModel.cs:14]: Could you add an example with expected value (with or without # char)

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs [hunk 8]
```diff
@@ -37,12 +33,23 @@ public class ChartModel : BaseElement
...
+        /// <summary>
+        /// Indicate if the legend must be displayed
+        /// </summary>
+        public bool ShowLegend { get; set; }
+
         /// <summary>
         /// Legend font family
         /// </summary>
-        public string FontFamilyLegend { get; set; }
+        public string FontFamilyLegend { get; set; } = "Arial";
```
Comment (mathieumack) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs:47]: Arial by default is mandatory ?
I prefer to not define default font families as you can inherit value from parent objects.

**Review by mathieumack — DISMISSED** [review]
Please check comments and answer or fix it
Thanks !

File: OpenXMLSDK.Engine/ReportEngine/DataContext/ContextModel.cs (hunk 3 above)
Comment (Alteoris) [reply OpenXMLSDK.Engine/ReportEngine/DataContext/ContextModel.cs:130]: I remove this code because it's not possible to set this color in the context

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs (hunk 8 above)
Comment (Alteoris) [reply OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartModel.cs]: Fixed ;)

**Review by HisCodeness — APPROVED** [review]
